pkg/kube/controllers/statefulset/statefulset_rollout_controller.go

```diff
@@ -0,0 +1,68 @@
+package statefulset
+
+import (
+	"context"
+
+	"k8s.io/api/apps/v1beta2"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"code.cloudfoundry.org/quarks-utils/pkg/config"
+	"code.cloudfoundry.org/quarks-utils/pkg/ctxlog"
+)
+
+// AddStatefulSetRollout creates a new statefulset rollout controller and adds it to the manager.
+// The purpose of this controller is to remove the partition of the statefulset if the canary succeeds.
+func AddStatefulSetRollout(ctx context.Context, config *config.Config, mgr manager.Manager) error {
+	ctx = ctxlog.NewContextWithRecorder(ctx, "statefulset-rollout-reconciler", mgr.GetEventRecorderFor("statefulset-rollout-recorder"))
+	r := NewStatefulSetRolloutReconciler(ctx, config, mgr)
+
+	// Create a new controller
+	c, err := controller.New("statefulset-rollout-controller", mgr, controller.Options{
+		Reconciler:              r,
+		MaxConcurrentReconciles: config.MaxQuarksStatefulSetWorkers,
+	})
+	if err != nil {
+		return errors.Wrap(err, "Adding StatefulSet rollout controller to manager failed.")
+	}
+
+	// Trigger when annotation is set
+	statefulSetPredicates := predicate.Funcs{
+		CreateFunc:  func(e event.CreateEvent) bool { return false },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+		GenericFunc: func(e event.GenericEvent) bool { return false },
+		UpdateFunc:  CheckUpdate,
```

Comment:
We use 'predicate logging' on all controllers to generate events and debug information.

Something like this: 
https://github.com/cloudfoundry-incubator/cf-operator/blob/master/pkg/kube/controllers/boshdeployment/deployment_controller.go#L62-L65

Reply:
We have added it...